src/markdown_parser.py

```diff
@@ -51,9 +51,22 @@ def urlreplace(matchobj: Match[str]) -> str:
 
         return re.sub(URL_REGEX, urlreplace, text)
 
+    def link(self, link, text=None, title=None):
+        # the parser may pass in `title`, but Asana's API does not allow the
+        # `title` attribute and is therefore ignored here.
+        # https://developers.asana.com/docs/rich-text
+
+        is_asana_vanity_link = link.startswith("https://app.asana.com") and text
+        if text is None:
+            text = link
+
+        asana_tags = 'data-asana-dynamic="false" ' if is_asana_vanity_link else ""
+        s = "<a " + asana_tags + 'href="' + self._safe_url(link) + '"'
+        return s + ">" + (text or link) + "</a>"
```

Comment:
I don't think it matters really since text is used like this (`text or link`).  I think the `text = link` line can be removed all together